zerver/views/streams.py

```diff
@@ -244,6 +245,46 @@ def remove_default_stream(
     return json_success()
 
 
+def send_change_stream_message_retention_days_notification(
+    user_profile: UserProfile, stream: Stream, old_value: Optional[int], new_value: Optional[int]
+) -> None:
+    sender = get_system_bot(settings.NOTIFICATION_BOT, user_profile.realm_id)
+    user_mention = f"@_**{user_profile.full_name}|{user_profile.id}**"
```

Comment:
We should do a prep commit to extract this code to a helper method `silent_mention_syntax_for_user(user_profile)` that we can call from all the places that are doing this sort of formatting by hand.  `zerver/lib/mention.py` might be a good home.